source/scripts/enum.lua

```diff
@@ -0,0 +1,26 @@
+module(...,package.seeall)

```

Comment:
I know this isn't your bit of code, but this line had always confused me as to what it exactly does and if it is needed

Reply:
I was waiting for someone to say something.  :)

It's old style LUA and it's how it was done "back then". I'll change it when I'm done reminiscing.  lolz

Reply:
I'm not happy with that line either.

Its used to set an environment for the containing variables and functions and returns them as a module / table.

Same as
```lua
local enum = {}
enum.foo = bar
return enum
```

Its just to not write enum everytime.
It has a big flaw though. For it to work it has to kinda inherite the global space which means all global variables and functions can be accessed through the module like `enum.print("Foobar")` which is a problem in my book.

Reply:
No problem. It confused me at first because we have now have a module called modules.